beacon_node/eth2_libp2p/src/discovery/mod.rs

```diff
@@ -53,6 +55,9 @@ const FIND_NODE_QUERY_CLOSEST_PEERS: usize = 16;
 /// The threshold for updating `min_ttl` on a connected peer.
 const DURATION_DIFFERENCE: Duration = Duration::from_millis(1);
 
+/// Interval duration for how often we should process queued queries.
+const QUERY_INTERVAL_DURATION: Duration = Duration::from_secs(2);
```

Comment:
Why is this necessary?

What was wrong with consuming the queue on demand?

Reply:
Did this mainly to improve the grouping of subnet queries. We are adding the subnet queries one by one into the `queued_queries` deque. The [`process_queue`](https://github.com/sigp/lighthouse/blob/332375c643eb5a4f8f7385a4b2e67c681bbfca49/beacon_node/eth2_libp2p/src/discovery/mod.rs#L436) function gets called rapidly every poll and may end up creating a `GroupedQuery` for just a single subnet every time it's called.

Giving a bit of a delay fills up `queued_queries` so that we get more subnets into a single `GroupedQuery`.
The delay could be much smaller though. 

I started modifying the discovery code so that we can add all the subnet queries we got at once, but I felt it changed too much in discovery for the scope of this PR and was better for a future PR.

Reply:
Did some more testing and realised consuming on demand is better since `queued_queries` get's populated faster than it get's consumed by the poll. Ignore the above comment. 
Will revert back to consuming queue on demand.